.github/workflows/ci-tests.yml

```diff
@@ -35,6 +35,13 @@ jobs:
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python_version }}
+        cache: pip
```

Comment:
this won't help it seems: https://github.com/actions/setup-python#caching-packages-dependencies

> The requirements file format allows for specifying dependency versions using logical operators (for example chardet>=3.0.4) or specifying dependencies without any versions. In this case the pip install -r requirements.txt command will always try to install the latest available package version.

that's how our `requirements.txt` is structured so according to the docs, the cache won't be used. we currently use `poetry export` to create it and there's no option to specify exact versions. (personally I like that, it _should_ check for updated packages without a change to `requirements.txt`)

Reply:
Huh, interesting, I can see it has been using the cache. Ok then:) Maybe only until there's one newer version available in between runs? Anyways, let's keep it.